packages/frontend-app-api/src/extensions/AppRoot.tsx

```diff
@@ -40,26 +47,43 @@ export const AppRoot = createExtension({
   attachTo: { id: 'app', input: 'root' },
   inputs: {
     signInPage: createExtensionInput(
-      {
-        component: createSignInPageExtension.componentDataRef,
-      },
+      { component: createSignInPageExtension.componentDataRef },
       { singleton: true, optional: true },
     ),
     children: createExtensionInput(
-      {
-        element: coreExtensionData.reactElement,
-      },
+      { element: coreExtensionData.reactElement },
       { singleton: true },
     ),
+    elements: createExtensionInput(
+      { element: coreExtensionData.reactElement },
+      { optional: true },
```

Comment:
No need to mark these as optional really, since they're not singletons

Reply:
Which is possibly a bit unfortunate - then you can't express "at least one". Alright, removing

Reply:
Yep I think it might make sense to flip it around to be `required` instead and have it apply to arrays too. So for it's the less common option